src/Microsoft.OData.Core/UriParser/SyntacticAst/CountSegmentToken.cs

```diff
@@ -0,0 +1,122 @@
+// </copyright>
+//---------------------------------------------------------------------
+
+#if ODATA_CLIENT
+namespace Microsoft.OData.Client.ALinq.UriParser
+#else
+namespace Microsoft.OData.UriParser
+#endif
+{
+    /// <summary>
+    /// Lexical token representing the $count segment in a path.
+    /// </summary>
+    public sealed class CountSegmentToken : PathToken
+    {
+        /// <summary>
+        /// The name of this token, which in this case is always "$count".
+        /// </summary>
+        private readonly string identifier;
+
+        /// <summary>
+        /// The instance to count on.
+        /// </summary>
+        private QueryToken nextToken;
+
+        /// <summary>
+        /// The token representing $filter.
+        /// </summary>
+        private QueryToken filterOption;
+
+        /// <summary>
+        /// The token representing $search.
+        /// </summary>
+        private QueryToken searchOption;
+
+        /// <summary>
+        /// Create a CountSegmentToken given the Identifier and the NextToken.
+        /// </summary>
+        /// <param name="identifier">The name of this token, which in this case is always "$count".</param>
+        /// <param name="nextToken">The instance to count on.</param>
+        public CountSegmentToken(string identifier, QueryToken nextToken)
+            :this(identifier, nextToken, null, null)
+        {
+        }
+
+        /// <summary>
+        /// Create a CountSegmentToken given the Identifier, NextToken, FilterOption (if any) and SearchOption (if any).
+        /// </summary>
+        /// <param name="identifier">The name of this token, which in this case is always "$count".</param>
+        /// <param name="nextToken">The instance to count on.</param>
+        /// <param name="filterOption">The <see cref="QueryToken"/> representing $filter.</param>
+        /// <param name="searchOption">The <see cref="QueryToken"/> representing $search. </param>
+        public CountSegmentToken(string identifier, QueryToken nextToken, QueryToken filterOption, QueryToken searchOption)
+        {
+            ExceptionUtils.CheckArgumentStringNotNullOrEmpty(identifier, "identifier");
+            ExceptionUtils.CheckArgumentNotNull(nextToken, "nextToken");
+
+            this.identifier = identifier;
+            this.nextToken = nextToken;
+            this.filterOption = filterOption;
+            this.searchOption = searchOption;
+        }
+
+        /// <summary>
+        /// The kind of the query token.
+        /// </summary>
+        public override QueryTokenKind Kind
+        {
+            get { return QueryTokenKind.CountSegment; }
+        }
+
+        /// <summary>
+        /// The name of this token, which in this case is always "$count".
+        /// </summary>
+        public override string Identifier
+        {
+            get { return this.identifier; }

```

Comment:
>this.identifier [](http://example.com/codeflow?start=25&length=15)

Just return ExpressionConstants.QueryOptionCount and remove the identifier private variable and constructor argument.

Reply:
Done